install/install.sh

```
@@ -0,0 +1,521 @@
+    
+    if [[ ! -f "$binary_file" ]]; then
+        print_error "Binary file not found: $binary_file"
+        return 1
+    fi
+    
+    if [[ ! -f "$hash_file" ]]; then
+        print_error "Hash file not found: $hash_file"
+        return 1
+    fi
+    
+    local calculated_hash=$(calculate_hash "$binary_file")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)